### PR TITLE
add dataframe of github hosseinmoein/DataFrame

### LINKED ISSUE #14024: [New Port Request] dataframe
Library name: dataframe

Library description: This is a C++ statistical library that provides an interface similar to Pandas package in Python

Source repository URL: https://github.com/hosseinmoein/DataFrame

### PR DESCRIPTION
**Pull request by shimondoodkin**
**Describe the pull request**

- What does your PR fix? Fixes
#14024 
#12364
#8640
#3369

- Which triplets are supported/not supported? Have you updated the CI baseline?
I built only with x64-windows. it is compiling.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes

### CODE REVIEW

**Comment by shimondoodkin** [comment]
not sure what to do with missing crt functions in uwp:
I will not develop further.

https://docs.microsoft.com/en-us/cpp/cppcx/crt-functions-not-supported-in-universal-windows-platform-apps?view=msvc-160

the error was:

```text

\HeteroVector.cc" "D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Vectors\HeteroView.cc" "D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Vectors\HeteroPtrView.cc" "D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\ThreadGranularity.cc" "D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc"
         HeteroVector.cc
         HeteroView.cc
         HeteroPtrView.cc
         ThreadGranularity.cc
         DateTime.cc
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\include\DataFrame/Utils/FixedSizeString.h(312,1): warning C4305: 'initializing': truncation from 'unsigned __int64' to 'hmdf::VirtualString::size_type' (compiling source file D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc) [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\include\DataFrame/Utils/FixedSizeString.h(312,29): warning C4309: 'initializing': truncation of constant value (compiling source file D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc) [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\include\DataFrame/Utils/FixedSizeString.h(315,39): warning C4244: '=': conversion from 'unsigned __int64' to 'hmdf::VirtualString::size_type', possible loss of data (compiling source file D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc) [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc(129,5): error C3861: '_tzset': identifier not found [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc(972,9): error C3861: '_tzset': identifier not found [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>D:\buildtrees\dataframe\src\9b26a72fa1-4a13f75ffc.clean\src\Utils\DateTime.cc(988,9): error C3861: '_tzset': identifier not found [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\XamlCompiler\Microsoft.Windows.UI.Xaml.Common.targets(486,5): error MSB4181: The "CompileXaml" task returned false but did not log an error. [D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj]
     4>Done Building Project "D:\buildtrees\dataframe\arm-uwp-dbg\DataFrame.vcxproj" (default targets) -- FAILED.
     3>Done Building Project "D:\buildtrees\dataframe\arm-uwp-dbg\ALL_BUILD.vcxproj" (default targets) -- FAILED.
     1>Done Building Project "D:\buildtrees\dataframe\arm-uwp-dbg\install.vcxproj" (default targets) -- FAILED.

Build FAILED.

```

**Comment by JackBoosY** [comment]
Because `_tzset` is not supported with uwp, see [documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/tzset?view=msvc-160).
So please add `Supports: !uwp` to the CONTROL file.

File: ports/dataframe/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    REF 106fb398a3a05a9d4055a1b00d1e3b9b26a72fa1
+    SHA512 43bd888312e16866f399e47adf7d0dc4fb4fd961063d3f53df4a69fb69f1ec95a1c294a4cefc4135f1acf6a4f27715fdcc71cb4bf69f2f72557290d9af439774
+    HEAD_REF master
+)
+
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:13]:
```suggestion
```

File: ports/dataframe/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    
+    OPTIONS
+        -DENABLE_TESTING:BOOL=OFF
+
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:16]:
```suggestion
```

File: ports/dataframe/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_fixup_cmake_targets(CONFIG_PATH cmake TARGET_PATH share/dataframe)
+else()
+    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake TARGET_PATH share/dataframe)
+endif()
+
+file( REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/cmake
+    ${CURRENT_PACKAGES_DIR}/debug/cmake
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:29]: This two directories should be removed by `vcpkg_fixup_cmake_targets`, can you confirm that?

File: ports/dataframe/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,35 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO hosseinmoein/DataFrame
+    REF 106fb398a3a05a9d4055a1b00d1e3b9b26a72fa1
+    SHA512 43bd888312e16866f399e47adf7d0dc4fb4fd961063d3f53df4a69fb69f1ec95a1c294a4cefc4135f1acf6a4f27715fdcc71cb4bf69f2f72557290d9af439774
+    HEAD_REF master
+)
+
+
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:9]:
```suggestion

```

File: ports/dataframe/CONTROL [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Source: dataframe
+Version: 1.12.0
+Description: This is a C++ statistical library that provides an interface similar to Pandas package in Python
```
Comment (JackBoosY) [inline ports/dataframe/CONTROL:3]:
```suggestion
Description: This is a C++ statistical library that provides an interface similar to Pandas package in Python
Homepage: https://github.com/hosseinmoein/DataFrame

```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/dataframe/portfile.cmake (hunk 3 above)
Comment (shimondoodkin) [reply ports/dataframe/portfile.cmake:25]: yes after vcpkg_fixup_cmake_targets 
seems not exists anymore but includes was still existed.

**Comment by shimondoodkin** [comment]
I made all the changes you asked for.

**Review by JackBoosY — APPROVED** [review]

File: ports/dataframe/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DENABLE_TESTING:BOOL=OFF
+)
+
+vcpkg_install_cmake()
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_fixup_cmake_targets(CONFIG_PATH cmake TARGET_PATH share/dataframe)
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:18]:
```suggestion
    vcpkg_fixup_cmake_targets(CONFIG_PATH cmake/DataFrame TARGET_PATH share/dataframe)

```

File: ports/dataframe/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+else()
+    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake TARGET_PATH share/dataframe)
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:20]:
```suggestion
    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/DataFrame TARGET_PATH share/dataframe)

```

File: ports/dataframe/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+endif()
+
```
Comment (JackBoosY) [inline ports/dataframe/portfile.cmake:22]:
```suggestion
vcpkg_fixup_pkgconfig()

```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**